python/requirements.txt

```diff
@@ -4,112 +4,65 @@
 #
 # In short, if you change it here, PLEASE also change it in setup.py.
 
+# You can obtain this list from the ray.egg-info/requires.txt
+
 ## setup.py install_requires
-aiohttp>=3.7
-aiosignal
-click >= 7.0, <= 8.0.4
-cloudpickle
+attrs
+click<=8.0.4,>=7.0
 filelock
-frozenlist
-gpustat >= 1.0.0
-grpcio >= 1.32.0; python_version < '3.10'
-grpcio >= 1.42.0; python_version >= '3.10'
 jsonschema
-msgpack >= 1.0.0, < 2.0.0
-numpy >= 1.16
-opencensus
-packaging; python_version >= '3.10'
-prometheus_client >= 0.7.1, < 0.14.0
-protobuf >= 3.15.3, != 3.19.5
-py-spy >= 0.2.0
-pydantic >= 1.8, < 1.10.0
+msgpack<2.0.0,>=1.0.0
+protobuf<4.0.0,>=3.15.35; platform_system != "Windows"
```

Comment:
What's the background of pinning to protobuf<4 when not on windows?

Reply:
Ah it seems we missed the relaxation of that constraint in https://github.com/ray-project/ray/commit/37c2b58bee37fdad945ee7e1ddf54b8586bed39a in this PR.
Looks like we can get rid of the custom logic and remove the pin to < 4.00

Reply:
Thanks for catching this. Fixing it here: https://github.com/ray-project/ray/pull/29584